Plugins/GitUIPluginInterfaces/PluginsPathScanner.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GitUIPluginInterfaces
+{
+    /// <summary>
+    /// A helper class for scanning for plugin files.
+    /// </summary>
+    public static class PluginsPathScanner
+    {
+        public static IEnumerable<FileInfo> GetFiles(params string[] pluginsPaths)
+        {
+            var result = Enumerable.Empty<FileInfo>();
+
+            foreach (string pluginsPath in pluginsPaths)
+            {
+                if (!string.IsNullOrEmpty(pluginsPath) && Directory.Exists(pluginsPath))
+                {
+                    DirectoryInfo directory = new DirectoryInfo(pluginsPath);
+
+                    result = Enumerable.Concat(result, directory.GetFiles("*.dll"));
```

Comment:
I would add a paragraph explaining why we are loading two levels of assemblies.
E.g. load plugins located in the folder to provide back compat, and load new style plugin...
Maybe even add a structure to help visualise
```
UserPlugins
  \OldPlugin1.dll
  \OldPlugin2.dll
  \OldPlugin2.config
  \NewPlugin1
      \Plugin1.dll
      \Plugin1.config
  \NewPlugin2
      \Plugin2.dll
      \Plugin2.config
```

Reply:
Ok